Improvement of `PeriodDemandSink` (#12)
EnergyModelsX/EnergyModelsFlex.jl: docs/src/nodes/sink/perioddemand.md

- **`penalty::Dict{Symbol,<:TimeProfile}`**:\
  The penalty dictionary is used for providing penalties for soft constraints to allow for both over and under delivering the demand.\
  It must include the fields `:surplus` and `:deficit`.
  In addition, it is crucial that the sum of both values in each demand period is larger than 0 to avoid an unconstrained model.

  !!! warning "Chosen values"
      The implementation is relative to the chosen `period_duration` (see below).

>       If the period duration is ``24``, then the cost is for the unsatisfied demand within the ``24`` demand period, multiplied with the probablity and the repititons within a strategic period.

Reviewer: ```suggestion
      If the period duration is ``24``, then the cost is for the unsatisfied demand within the ``24`` demand period, multiplied with the probability and the repetitons within a strategic period.
```

[`AbstractPeriodDemandSink`](@ref EnergyModelsFlex.AbstractPeriodDemandSink)s require additional fields to specify both the periods and their respective demands:

- **`period_duration::TimeProfile`**:\
  Defines what the total duration of a single demand period.\
  For instance, if the duration of 1 of the operation time structure is 1 hour and `period_duration = 24`, then each demand period spans one day.
  The demand of this node (for a given day, see below) must then be filled on a daily basis, without any restrictions on *when* during the day the demand must be filled given the available capacity.\
  Due to a constructor, it can either be specified as number (the same duration in all demand periods), as a vector (varying duration of each demand period), or as a time profile (*e.g.*, varying period durations due to varying operational time structures).
  It cannot be specified as `OperationalProfile`.

- **`period_demand::TimeProfile`**:\
  The total demand to be met during each demand period.
  The length of this vector should match the number of periods (*e.g.*, days) in the time structure.

>   If the time structure represents on year with hourly resolution and the demand periods correspond to a day, this vector must then have 365 elements.

Reviewer: ```suggestion
  If the time structure represents one year with hourly resolution and the demand periods correspond to a day, this vector must then have 365 elements.
```

>   If the time structure represents on year with hourly resolution and the demand periods correspond to a day, this vector must then have 365 elements.

Reviewer: Refrase from `vector` to be consistent here.

>   It is best to utilize the [`PartitionProfile`]() type if the demand is varying.

Reviewer: Link is empty.

>   If it is constent, you can also utilize [`StrategicProfile`][@extref TimeStruct.StrategicProfile], [`RepresentativeProfile`][@extref TimeStruct.RepresentativeProfile], or [`ScenarioProfile`][@extref TimeStruct.ScenarioProfile], depending on your chosen time structure.

Reviewer: ```suggestion
  If it is constant, you can also utilize [`StrategicProfile`][@extref TimeStruct.StrategicProfile], [`RepresentativeProfile`][@extref TimeStruct.RepresentativeProfile], or [`ScenarioProfile`][@extref TimeStruct.ScenarioProfile], depending on your chosen time structure.
```

Reviewer: or rather "constant on the operational level"?
